conformance/lang/expressions/list-constructor-expr/spread_operator.balt

```diff
@@ -0,0 +1,728 @@
+    io:println(v4); // @output [1,2,3,4]
+    io:println(v5); // @output [1,2,3,4,1,1]
+}
+
+Test-Case: output
+Description: Test spread operator with type cast expr.
+Labels: array-type, list-constructor-expr, spread-operator, tuple-type, union-type, type-cast-expr
+
+function init() {
+    int[*]|string[*] a1 = [1, 2];
+    int[*]|string[*] a2 = [3, 4];
+    [(int|string)...] v0 = [...<int[2]>a1, ...<int[2]>a2];
+    [int, (int|string)...] v1 = [...<int[2]>a1, ...<int[2]>a2];
+    [int, int, (int|string)...] v2 = [...<int[2]>a1, ...<int[2]>a2];
+    [int, any, (int|string)...] v3 = [...<int[2]>a1, ...<int[2]>a2];
+    [int, anydata, (int|string)...] v4 = [...<int[2]>a1, ...<int[2]>a2];
+    [int, (int|string)...] v5 = [...<int[2]>a1, ...<int[2]>a2, ...[1, 1]];
+    io:println(v0); // @output [1,2,3,4]
+    io:println(v1); // @output [1,2,3,4]
+    io:println(v2); // @output [1,2,3,4]
+    io:println(v3); // @output [1,2,3,4]
+    io:println(v4); // @output [1,2,3,4]
+    io:println(v5); // @output [1,2,3,4,1,1]
+}
+
+Test-Case: output
+Description: Test spread operator with tuple type.
+Labels: list-constructor-expr, spread-operator, tuple-type, union-type
+
+function init() {
+    [int, string] a1 = [1, "x"];
+    [int...] a2 = [1, 2];
+    [(int|string)...] v0 = [...a1, ...a2];
+    [int, (int|string)...] v1 = [...a1, ...a2];
+    [int, string, (int|string)...] v2 = [...a1, ...a2];
+    [int, any, (int|string)...] v3 = [...a1, ...a2];
+    [int, anydata, (int|string)...] v4 = [...a1, ...a2];
+    [int, (int|string)...] v5 = [...a1, ...a2, ...[1, 1]];
+    io:println(v0); // @output [1,"x",3,4]
+    io:println(v1); // @output [1,"x",3,4]
+    io:println(v2); // @output [1,"x",3,4]
+    io:println(v3); // @output [1,"x",3,4]
+    io:println(v4); // @output [1,"x",3,4]
+    io:println(v5); // @output [1,"x",3,4,1,1]
+}
+
+Test-Case: output
+Description: Test spread operator with tuple type.
+Labels: list-constructor-expr, spread-operator, tuple-type, union-type, never-type
+
+function init() {
+    [int, string, never...] a1 = [1, "x"];
+    [int...] a2 = [1, 2];
+    [never...] a3 = [];
+    [(int|string)...] v0 = [...a1, ...a2, ...a3];
+    [int, (int|string)...] v1 = [...a1, ...a2, ...a3];
+    [int, string, (int|string)...] v2 = [...a1, ...a2, ...a3];
+    [int, any, (int|string)...] v3 = [...a1, ...a2, ...a3];
+    [int, anydata, (int|string)...] v4 = [...a1, ...a2, ...a3];
+    [int, (int|string)...] v5 = [...a1, ...a2, ...a3, ...[1, 1]];
+    io:println(v0); // @output [1,"x",1,2]
+    io:println(v1); // @output [1,"x",1,2]
+    io:println(v2); // @output [1,"x",1,2]
+    io:println(v3); // @output [1,"x",1,2]
+    io:println(v4); // @output [1,"x",1,2]
+    io:println(v5); // @output [1,"x",1,2,1,1]
+}
+
+Test-Case: output
+Description:  Test spread operator without contextually expected type.
+Labels: list-constructor-expr, spread-operator, tuple-type, var, never-type
+
+function init() {
+    [int, string, never...] a1 = [1, "x"];
```

Comment:
never rest descriptor is a special case. Better add those tests separately. If that is not considered in the implementation 
we have to disable the "spread operator without contextually expected type" test.